docs/src/main/sphinx/connector/pinot.rst

```diff
@@ -74,6 +74,10 @@ Property name                                             Required   Description
 ``pinot.aggregation-pushdown.enabled``                    No         Push down aggregation queries, default is ``true``.
 ``pinot.count-distinct-pushdown.enabled``                 No         Push down count distinct queries to Pinot, default is ``true``.
 ``pinot.target-segment-page-size``                        No         Max allowed page size for segment query, default is ``1MB``.
+``pinot.extra-http-headers``                              No         Extra headers when sending HTTP based pinot requests to Pinot controller/broker.
```

Comment:
I have little concern to have such property that we can't test easily. For instance, when users try to set this property, face some issue and ask in Slack or somewhere, it' basically hard to help them in my opinion. Could you share more details about "customization, tracing, or tag information" and describe why dedicate properties don't work?

Reply:
One use case here is that we want to add extra headers to let Pinot side track the sources and tags of which presto cluster it queries. Another use case is to allow another auth mechanism in Pinot, e.g. using an API key.

Reply:
Thanks.

> One use case here is that we want to add extra headers to let Pinot side track the sources and tags of which presto cluster it queries. 

This doesn't explain why dedicate properties don't work. Do you mean Pinot doesn't have prepared configuration for sources and tags?

> Another use case is to allow another auth mechanism in Pinot, e.g. using an API key.

We should add dedicated property when we want to support new auth mechanism.

Reply:
This is the catch all solution for  all the other cases. 

During dev/staging process, it may not worth to patch Trino, config changes will be much simpler.

Reply:
If the purpose is for dev/staging process, the current property name is incorrect and we should hide from documentation in my opinion.

Reply:
No. It's not for dev purpose. I'm just listing all the scenarios this thing can be useful. 

This is the catch all solution to allow users to customize what to pass to Pinot side.

From users perspective, those Pinot side customization may not be put into open source code. And there will always be delay from implementing features in Pinot then enable it in Trino.